sync_metadata.go

```diff
@@ -0,0 +1,14 @@
+// Copyright 2024-Present Couchbase, Inc.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+// in that file, in accordance with the Business Source License, use of this
+// software will be governed by the Apache License, Version 2.0, included in
+// the file licenses/APL2.txt.
+
+package sgbucket
+
+const (
+	SyncDocPrefix = "_sync:"                // SyncDocPrefix is a document prefix for all sync metadata
```

Comment:
I don't feel like these are a natural fit for sg-bucket, and I don't like splitting the prefix definitions into two places (here, and constants_syncdocs.go).  I think I'd prefer to redefine these in rosmar if that's what's needed to avoid circular dependencies.